cmake/projects/OpenSSL/schemes/url_sha1_openssl.cmake.in

```diff
@@ -50,11 +52,17 @@ if(ANDROID)
     hunter_user_error("Could not find android architecture. Please set ANDROID_ARCH_NAME in your toolchain or use CMake 3.7+")
   endif()
   set(configure_command
-      # Ignored. Prevents script from checking host uname
-      RELEASE=2.6.37
-      SYSTEM=android
-      ARCH=${ANDROID_SSL_ARCH}
-      ${configure_command})
+      ${configure_command} ${CMAKE_C_FLAGS})
+
+  # Latest OpenSSL (1.1.1) still does not have correct android targets, so set generic with our CFLAGS
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "arm64" _is_arm64)
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "mips64" _is_mips64)
+  string(COMPARE EQUAL "${ANDROID_SSL_ARCH}" "x86_64" _is_x8664)
+  if (_is_arm64 OR _is_mips64 OR _is_x8664)
+    set(configure_opts "linux-generic64")
```

Comment:
> Is this what you want?

As I've already mentioned the approach you're using is completely different from previous approach. So I want to have something to show in case the next patch will comes from another user saying that your approach is not working in his configuration.

> The reason I am using linux-generic32/64 is there is no proper way to build it with our CFLAGS or Clang.

"This configuration works for me" is not an argument.